scripts/o3de/o3de/gem_properties.py

```diff
@@ -85,24 +88,26 @@ def edit_gem_props(gem_path: pathlib.Path = None,
                          f' characters, and start with a letter.  {new_name}')
             return 1
         update_key_dict['gem_name'] = new_name
-    if new_display:
+    if new_display is not None:
         update_key_dict['display_name'] = new_display
-    if new_origin:
+    if new_origin is not None:
         update_key_dict['origin'] = new_origin
-    if new_type:
+    if new_type is not None:
         update_key_dict['type'] = new_type
-    if new_summary:
+    if new_summary is not None:
         update_key_dict['summary'] = new_summary
-    if new_icon:
+    if new_icon is not None:
         update_key_dict['icon_path'] = new_icon
-    if new_requirements:
+    if new_requirements is not None:
         update_key_dict['requirements'] = new_requirements
-    if new_documentation_url:
+    if new_documentation_url is not None:
         update_key_dict['documentation_url'] = new_documentation_url
-    if new_license:
+    if new_license is not None:
         update_key_dict['license'] = new_license
-    if new_license_url:
+    if new_license_url is not None:
         update_key_dict['license_url'] = new_license_url
+    if new_repo_uri is not None:
```

Comment:
This could use python comprehension if "expression" as well

```python
update_key_dict['display_name'] = new_display if isinstance(new_display, str) else ""
```

At the very least a better check is to make sure the instance of the variable is `str` instead of checking against it not being `None`

Reply:
After speaking with AMZN-alexpete, it looks like using the comprehension suggested here creates a possible edge case from the CLI side. 

If someone tries to call the edit_gem_props function like this:
`edit_gem_props(gem_name="blah", new_display="new!")`
with only the intention to edit `gem_name` and `new_display`, then they end up unintentionally clearing every other field.

Reply:
most likely I plan on reverting back this change

Reply:
That works for me.